core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/join.kt

```diff
@@ -116,10 +118,20 @@ public enum class JoinType {
     Left, // all data from left data frame, nulls for mismatches in right data frame
     Right, // all data from right data frame, nulls for mismatches in left data frame
     Inner, // only matched data from right and left data frame
+    Filter, // only matched data from left data frame
     Full, // all data from left and from right data frame, nulls for any mismatches
     Exclude // mismatched rows from left data frame
 }
 
+internal val JoinType.addNewColumns: Boolean
+    get() {
```

Comment:
also here is an expression body probably cleaner, especially since you use the same name as the getter :)

Reply:
done